pkg/security/probe/tags_resolver.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build linux
+
+package probe
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/security/config"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/tagger/remote"
+)
+
+// Tagger defines a Tagger for the Tags Resolver
+type Tagger interface {
+	Init() error
+	Stop() error
+	Tag(entity string, cardinality collectors.TagCardinality) ([]string, error)
+}
+
+type nullTagger struct{}
+
+func (n *nullTagger) Init() error {
+	return nil
+}
+
+func (n *nullTagger) Stop() error {
+	return nil
+}
+
+func (n *nullTagger) Tag(entity string, cardinality collectors.TagCardinality) ([]string, error) {
+	return nil, nil
+}
+
+// MountResolver represents a cache for mountpoints and the corresponding file systems
+type TagsResolver struct {
+	probe *Probe
+
+	tagger Tagger
+}
+
+// Start the resolver
+func (t *TagsResolver) Start(ctx context.Context) error {
+	if err := t.tagger.Init(); err != nil {
+		return err
+	}
+
+	go func() {
+		<-ctx.Done()
+		t.tagger.Stop()
+	}()
+
+	return nil
+}
+
+// ResolverTags returns the tags for the given id
+func (t *TagsResolver) Resolve(id string) []string {
```

Comment:
We already discussed this on Slack, but we should probably remove the `ResolveLabels` method of the `ContainerResolver` :)